runtime/queries/rust/highlights.scm

```diff
@@ -1,95 +1,42 @@
-; Identifier conventions
+; Bottom-up approach, granular to broad.
 
+; Guaranteed Global
 
-; Assume all-caps names are constants
-((identifier) @constant
- (#match? @constant "^[A-Z][A-Z\\d_]+$'"))
-
-; Assume other uppercase names are enum constructors
-((identifier) @constructor
- (#match? @constructor "^[A-Z]"))
-
-; Assume that uppercase names in paths are types
-(mod_item
- name: (identifier) @namespace)
-(scoped_identifier
-  path: (identifier) @namespace)
-(scoped_identifier
- (scoped_identifier
-  name: (identifier) @namespace))
-(scoped_type_identifier
-  path: (identifier) @namespace)
-(scoped_type_identifier
- (scoped_identifier
-  name: (identifier) @namespace))
-
-((scoped_identifier
-  path: (identifier) @type)
- (#match? @type "^[A-Z]"))
-((scoped_identifier
-  path: (scoped_identifier
-    name: (identifier) @type))
- (#match? @type "^[A-Z]"))
-
-; Namespaces
-
-(crate) @namespace
-(extern_crate_declaration
-    (crate)
-    name: (identifier) @namespace)
-(scoped_use_list
-  path: (identifier) @namespace)
-(scoped_use_list
-  path: (scoped_identifier
-            (identifier) @namespace))
-(use_list (scoped_identifier (identifier) @namespace . (_)))
-
-; Function calls
-
-(call_expression
-  function: (identifier) @function)
-(call_expression
-  function: (field_expression
-    field: (field_identifier) @function.method))
-(call_expression
-  function: (scoped_identifier
-    "::"
-    name: (identifier) @function))
-
-(generic_function
-  function: (identifier) @function)
-(generic_function
-  function: (scoped_identifier
-    name: (identifier) @function))
-(generic_function
-  function: (field_expression
-    field: (field_identifier) @function.method))
+"$" @punctuation.delimiter
```

Comment:
I think there's a reason for the old ordering. Won't this now match `$` in `$crate`?